tests/odbc/test-odbc-oracle.cpp

```diff
@@ -0,0 +1,301 @@
+{
+    table_creator_three(soci::session & sql)
+        : table_creator_base(sql)
+    {
+        sql << "create table soci_test(name varchar(100) not null, "
+            "phone varchar(15))";
+    }
+};
+
+struct table_creator_for_get_affected_rows : table_creator_base
+{
+    table_creator_for_get_affected_rows(soci::session & sql)
+        : table_creator_base(sql)
+    {
+        sql << "create table soci_test(val integer)";
+    }
+};
+
+struct table_creator_for_xml : table_creator_base
+{
+    table_creator_for_xml(soci::session& sql)
+        : table_creator_base(sql)
+    {
+        sql << "create table soci_test(id integer, x xmltype)";
+    }
+};
+
+struct table_creator_for_clob : table_creator_base
+{
+    table_creator_for_clob(soci::session& sql)
+        : table_creator_base(sql)
+    {
+        sql << "create table soci_test(id integer, s long)";
+    }
+};
+
+//
+// Support for SOCI Common Tests
+//
+
+class test_context : public test_context_base
+{
+public:
+    test_context(backend_factory const &backEnd,
+                std::string const &connectString)
+        : test_context_base(backEnd, connectString),
+          m_verDriver(get_driver_version())
+    {
+        std::cout << "Using ODBC driver version " << m_verDriver << "\n";
+    }
+
+    table_creator_base * table_creator_1(soci::session& s) const SOCI_OVERRIDE
+    {
+        return new table_creator_one(s);
+    }
+
+    table_creator_base * table_creator_2(soci::session& s) const SOCI_OVERRIDE
+    {
+        return new table_creator_two(s);
+    }
+
+    table_creator_base * table_creator_3(soci::session& s) const SOCI_OVERRIDE
+    {
+        return new table_creator_three(s);
+    }
+
+    table_creator_base * table_creator_4(soci::session& s) const SOCI_OVERRIDE
+    {
+        return new table_creator_for_get_affected_rows(s);
+    }
+
+    table_creator_base* table_creator_xml(soci::session& ) const SOCI_OVERRIDE
+    {
+        // return new table_creator_for_xml(s);
+        return NULL;
+    }
+
+    table_creator_base* table_creator_clob(soci::session& ) const SOCI_OVERRIDE
+    {
+        // return new table_creator_for_clob(s);
+        return NULL;
+    }
+
+    bool has_real_xml_support() const SOCI_OVERRIDE
+    {
+        return true;
+    }
+
+    std::string to_date_time(std::string const &datdt_string) const SOCI_OVERRIDE
+    {
+        return "to_date(\'" + datdt_string + "\', \'yyyy-mm-dd hh24:mi:ss\')";
+    }
+
+    bool has_fp_bug() const SOCI_OVERRIDE
+    {
+        // The bug with using insufficiently many digits for double values was
+        // only fixed in 9.03.0400 version of the ODBC driver (see commit
+        // a5fed2338b59ae16a2d3a8d2744b084949684775 in its repository), so we
+        // need to check for its version here.
+        //
+        // Be pessimistic if we failed to retrieve the version at all.
+        return !m_verDriver.is_initialized() || m_verDriver < odbc_version(9, 3, 400);
```

Comment:
It doesn't make sense to compare ODBC driver version with 9.03.0400 used for PstgresSQL driver.